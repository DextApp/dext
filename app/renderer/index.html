<!doctype html>
<html>
  <head>
    <title>Dext</title>
    <style type="text/css">
      body,
      html {
        margin: 0;
        padding: 0;
        background-color: transparent;
      }
      #app {
        background-color: transparent;
      }
    </style>
  </head>

  <body>

    <div id="app"></div>

    <script type="text/javascript" src="lib/bundle.js"></script>

  </body>
</html>
