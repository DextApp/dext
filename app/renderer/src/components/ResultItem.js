import React, { PropTypes } from 'react';
import { style, compose, hover } from 'glamor';
import Icon from './Icon';
import { ResultItemSchema, ThemeSchema } from '../schema';

const activeStyle = {
  backgroundColor: '#3f93fe',
  color: '#ffffff',
};

const baseHover = hover(activeStyle);

const base = compose(
  style({
    display: 'flex',
    color: '#333',
    boxSizing: 'border-box',
    height: 60,
    margin: 0,
    padding: 10,
    width: '100%',
  }),
  baseHover
);

const icon = style({
  flex: 'content',
  boxSizing: 'border-box',
  marginRight: 10,
});

const details = style({
  flex: 'flex-grow',
  boxSizing: 'border-box',
});

const title = style({
  display: '-webkit-box',
  boxSizing: 'border-box',
  overflow: 'hidden',
  fontSize: 16,
  fontWeight: 'normal',
  margin: 0,
  WebkitBoxOrient: 'vertical',
  WebkitLineClamp: 1,
});

const subtitle = style({
  display: '-webkit-box',
  boxSizing: 'border-box',
  overflow: 'hidden',
  fontSize: 12,
  fontWeight: 'normal',
  margin: 0,
  WebkitBoxOrient: 'vertical',
  WebkitLineClamp: 1,
});

const ResultItem = ({ theme, selected, item, isAltMod, isSuperMod, onDoubleClick }) => {
  const themeBase = style(theme.result || {});
  const themeHover = hover(theme.resultActive || {});

  const themeSelected = selected
    ? compose(
      style(activeStyle),
      style(theme.resultActive || {})
    )
    : {};

  // apply modifiers if necessary
  const itemSubtitle = (isSuperMod && item.mods && item.mods.cmd && item.mods.cmd.subtitle)
    || (isAltMod && item.mods && item.mods.alt && item.mods.alt.subtitle)
    || item.subtitle;

  return (
    <li {...compose(base, themeBase, themeHover, themeSelected)} onDoubleClick={onDoubleClick}>
      <div {...icon}>
        <Icon icon={item.icon} />
      </div>
      <div {...details}>
        <h2 {...compose(title, theme.resultTitle || {})}>{item.title}</h2>
        <h3 {...compose(subtitle, theme.resultSubtitle || {})}>{itemSubtitle}</h3>
      </div>
    </li>
  );
};

ResultItem.propTypes = {
  theme: ThemeSchema,
  // item prop should follow the Alfred workflow script filter JSON format
  // https://www.alfredapp.com/help/workflows/inputs/script-filter/json/
  item: ResultItemSchema,
  selected: PropTypes.bool,
  isAltMod: PropTypes.bool,
  isSuperMod: PropTypes.bool,
  onDoubleClick: PropTypes.func,
};

export default ResultItem;
