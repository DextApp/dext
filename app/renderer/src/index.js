/* global document */
import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import DevTools from './components/DevTools';
import AppContainer from './containers/AppContainer';
import configureStore from './store';

const store = configureStore();

// load the dev tools if necessary
let devTools = '';
if (process.env.NODE_ENV === 'development') {
  devTools = <DevTools />;
}

const DextApp = () => (
  <Provider store={store}>
    <div>
      <AppContainer />
      {devTools}
    </div>
  </Provider>
);

const rootNode = document.getElementById('app');

ReactDOM.render(<DextApp />, rootNode);
