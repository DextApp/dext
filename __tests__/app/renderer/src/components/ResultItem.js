import React from 'react';
import renderer from 'react-test-renderer';
import ResultItem from '../../../../../app/renderer/src/components/ResultItem';

describe('<ResultItem /> component', () => {
  let theme = {};

  beforeEach(() => {
    theme = {};
  });

  it('should render a result item component with a text icon', () => {
    const item = {
      title: 'Text Item',
      subtitle: 'Text Subtitle',
      icon: {
        type: 'text',
        letter: 'T',
      },
    };
    const component = renderer.create(
      <ResultItem theme={theme} item={item} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a result item component with a file icon', () => {
    const item = {
      title: 'File Title',
      subtitle: 'File Subtitle',
      icon: {
        type: 'file',
        path: './icon.png',
      },
    };
    const component = renderer.create(
      <ResultItem theme={theme} item={item} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a result item component in a selected state', () => {
    const item = {
      title: 'File Title',
      subtitle: 'File Subtitle',
      icon: {
        type: 'file',
        path: './icon.png',
      },
    };
    const component = renderer.create(
      <ResultItem theme={theme} item={item} selected />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
