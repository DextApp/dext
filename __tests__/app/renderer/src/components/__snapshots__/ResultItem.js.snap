exports[`<ResultItem /> component should render a result item component in a selected state 1`] = `
<li
  data-css-1j2x6j7=""
  onDoubleClick={undefined}>
  <div
    data-css-1ch5118="">
    <img
      data-css-1vehqxh=""
      role="presentation"
      src="./icon.png" />
  </div>
  <div
    data-css-d67j44="">
    <h2
      data-css-13rh4x1="">
      File Title
    </h2>
    <h3
      data-css-1rd6wq3="">
      File Subtitle
    </h3>
  </div>
</li>
`;

exports[`<ResultItem /> component should render a result item component with a file icon 1`] = `
<li
  data-css-wgw6d2=""
  onDoubleClick={undefined}>
  <div
    data-css-1ch5118="">
    <img
      data-css-1vehqxh=""
      role="presentation"
      src="./icon.png" />
  </div>
  <div
    data-css-d67j44="">
    <h2
      data-css-13rh4x1="">
      File Title
    </h2>
    <h3
      data-css-1rd6wq3="">
      File Subtitle
    </h3>
  </div>
</li>
`;

exports[`<ResultItem /> component should render a result item component with a text icon 1`] = `
<li
  data-css-wgw6d2=""
  onDoubleClick={undefined}>
  <div
    data-css-1ch5118="">
    <span
      data-css-1ezhq81=""
      role="presentation">
      T
    </span>
  </div>
  <div
    data-css-d67j44="">
    <h2
      data-css-13rh4x1="">
      Text Item
    </h2>
    <h3
      data-css-1rd6wq3="">
      Text Subtitle
    </h3>
  </div>
</li>
`;
